foxpuppet/windows/browser/bookmarks/bookmark.py

```diff
@@ -0,0 +1,213 @@
+
+                        if bookmark_data.get("keyword"):
+                            keyword = bookmark_data["keyword"] or ""
+                            self.actions.send_keys(keyword + Keys.TAB).perform()
+
+                    self.actions.send_keys(
+                        Keys.TAB, Keys.TAB, Keys.TAB, Keys.ENTER
+                    ).perform()
+                    if (
+                        folder := self.selenium.find_element(
+                            *BookmarkLocators.BOOKMARK_FOLDER
+                        )
+                    ) is not None:
+                        folder.click()
+                        self.selenium.switch_to.frame(folder)
+                        self.actions.send_keys(Keys.TAB, Keys.ENTER).perform()
+
+    def retrieve_bookmark(self, label: str) -> bool:
+        """
+        Check if a bookmark with the given label exists.
+
+        Args:
+            label (str): The name of the bookmark to search for.
+        """
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            self.selenium.find_element(*BookmarkLocators.PANEL_MENU).click()
+            self.selenium.find_element(*BookmarkLocators.PANEL_BOOKMARK_MENU).click()
+            panel_bookmarks = self.selenium.find_element(
+                *BookmarkLocators.PANEL_BOOKMARK_TOOLBAR
+            )
+            menu_items = panel_bookmarks.find_elements(
+                By.CSS_SELECTOR, "toolbarbutton.bookmark-item"
+            )
+            if any(
+                label.lower() in item_label.lower()
+                for item in menu_items
+                if (item_label := item.get_attribute("label"))
+            ):
+                return True
+        return False
+
+    def delete_bookmark(
+        self, label: Optional[str] = None, is_detailed: bool = False
+    ) -> None:
+        """
+        Delete a bookmark using either quick delete (star button) or detailed menu approach.
+
+        Args:
+            detailed (bool, optional): Whether to use detailed menu approach. Defaults to False.
+            label (str, optional): Label of the bookmark to delete when using detailed approach.
+                Required when detailed is True.
+
+        Returns:
+            bool: True if bookmark was successfully deleted (always True for detailed approach)
+        """
+        with self.selenium.context(self.selenium.CONTEXT_CHROME):
+            if not is_detailed:
+                star_button_image = self.selenium.find_element(
+                    *BookmarkLocators.STAR_BUTTON_IMAGE
+                )
+                if (
+                    star_button_image
+                    and star_button_image.get_attribute("starred") == "true"
+                ):
+                    self.selenium.find_element(*BookmarkLocators.STAR_BUTTON).click()
+                    self.selenium.find_element(*BookmarkLocators.REMOVE_BUTTON).click()
+            else:
+                self.actions.context_click(
```

Comment:
For operations like this where you have just 1 condition to check you can just have an `if` with the expression you want to evaluate ending with a `return` and then outside of that `if` block you can add the actions that will be done if the expression evaluates opposing to what you wanted.

So something like
```
if True:
    code here if true
    return
code here if false
```

Does that make sense?